src/caf/core/data_structures.py

```diff
@@ -684,8 +870,564 @@ def aggregate(self, segs: list[str]):
+
+        return DVector(
+            import_data=new_data,
+            segmentation=new_seg,
+            zoning_system=self.zoning_system,
+            time_format=self.time_format,
+            low_memory=self.low_memory,
+            val_col=self.val_col,
+            cut_read=True,
         )
 
+    def calc_rmse(self, targets: list[IpfTarget]):
+        """
+        Calculate the rmse relative to a set of targets.
+
+        Parameters
+        ----------
+        targets: list[IpfTarget]
+            The targets to calc rmse relative to.
+        """
+        mse = 0
+        for target in targets:
+            check = self.copy()
+            if target.zone_translation is not None:
+                check = self.translate_zoning(
+                    target.data.zoning_system,
+                    trans_vector=target.zone_translation,
+                    _bypass_validation=True,
+                )
+            if target.segment_translations is not None:
+                for seg in target.data.segmentation - self.segmentation:
+                    seg = seg.name
+                    if target.segment_translations[seg] in self.segmentation.names:
+                        lower_seg = self.segmentation.get_segment(
+                            target.segment_translations[seg]
+                        )
+                        check = check.translate_segment(
+                            from_seg=lower_seg.name, to_seg=seg, _bypass_validation=True
+                        )
+                    else:
+                        raise ValueError("No translation defined for this segment.")
+            diff = (
+                check.aggregate(target.data.segmentation, _bypass_validation=True).__sub__(
+                    target.data, _bypass_validation=True
+                )
+            ) ** 2
+            mse += diff.sum() / len(target.data)
+        return mse**0.5
+
+    def validate_ipf_targets(self, targets: list[IpfTarget]):
+        """
+        Check targets for ipf will work, raises errors if not.
+
+        Parameters
+        ----------
+        targets: list[IpfTarget]
+            List of IPF targets to validate.
+
+        Returns
+        -------
+        Input targets, with zone translations added in if relevant.
+        """
+        target_sum = 0
+        for position, target in enumerate(targets):
+            # Check targets sum to the same, or they can't converge. Potentially could allow
+            # IPF for non-agreeing targets to get as close as possible.
+            zoning_diff = False
+            if target_sum == 0:
+                target_sum = target.data.sum()
+            else:
+                # TODO don't hard code this
+                if not math.isclose(target_sum, target.data.sum(), abs_tol=target_sum / 1e5):
+                    raise ValueError(
+                        "Input target DVectors do not have consistent "
+                        "sums, so ipf will fail."
+                    )
```

Comment:
Error should detail which IPD target is has caused it, just adding position to the error is probably enough